x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { EuiFieldText, EuiFormRow, EuiText } from '@elastic/eui';
+import { cloneDeep } from 'lodash';
+import { policyConfig } from '../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from './policy_hooks';
+
+function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
+  let newPolicyConfig = obj;
+  for (let i = 0; i < path.length - 1; i++) {
+    if (!newPolicyConfig[path[i]]) {
+      newPolicyConfig[path[i]] = {} as Record<string, unknown>;
+    }
+    newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
+  }
+  newPolicyConfig[path[path.length - 1]] = value;
+}
+
+function getValue(obj: Record<string, unknown>, path: string[]) {
+  let currentPolicyConfig = obj;
+
+  for (let i = 0; i < path.length - 1; i++) {
+    if (currentPolicyConfig[path[i]]) {
+      currentPolicyConfig = currentPolicyConfig[path[i]] as Record<string, unknown>;
+    } else {
+      return undefined;
+    }
+  }
+  return currentPolicyConfig[path[path.length - 1]];
+}
+
+export const PolicyAdvanced = React.memo(
+  ({
+    configPath,
+    firstSupportedVersion,
+    lastSupportedVersion,
+  }: {
+    configPath: string[];
+    firstSupportedVersion: string;
+    lastSupportedVersion?: string;
+  }) => {
+    const dispatch = useDispatch();
+    const policyDetailsConfig = usePolicyDetailsSelector(policyConfig);
+    const onChange = useCallback(
+      (event) => {
+        if (policyDetailsConfig) {
+          const newPayload = cloneDeep(policyDetailsConfig);
+          setValue(
+            (newPayload as unknown) as Record<string, unknown>,
+            event.target.value,
+            configPath
+          );
+          dispatch({
+            type: 'userChangedPolicyConfig',
+            payload: { policyConfig: newPayload },
+          });
+        }
+      },
+      [dispatch, policyDetailsConfig, configPath]
+    );
+
+    const value =
+      policyDetailsConfig &&
+      getValue((policyDetailsConfig as unknown) as Record<string, unknown>, configPath);
+
+    return (
+      <>
```

Comment:
we should wrap this in an `EuiPanel` component to match the mock.

https://elastic.github.io/eui/#/layout/panel

Reply:
f44601c417470f867e7aff2f38871e449f948725